src/setuptools_scm/version.py

```diff
@@ -168,7 +201,8 @@ def format_next_version(self, guess_next, fmt="{guessed}.dev{distance}", **kw):
 def _parse_tag(tag, preformatted, config):
     if preformatted:
         return tag
-    if not isinstance(tag, Version):
+    version_cls = _get_version_cls(config)
```

Comment:
by making the version class a mandatory attribute of the config this will no longer be necessary

Reply:
ok